src/main/java/com/google/sps/ImageTextDectector.java

```diff
@@ -16,45 +16,116 @@
  * list items and detect text from it. 
- * 2) It then uses an algorithm to create shopping sentences (queries)
- * from the text and their position. 
- * 3) This list of queries is returned to the Servlet from the method 'extractShoppingList'.
+ * 2) It then uses an algorithm to extract shopping sentences (queries)
+ * from the text based on their position (y coordinates). 
+ * 3) This list of queries is returned to the Servlet from the {@link #extractShoppingList(byte[])}.
  */
 public class ImageTextDectector {
 
-  TextDetectionAPI textDetectionAPI;
+  // This variable is used to invoke interface's implimentation for real CloudVisionAPI calls and fake calls from tests.
+  private TextDetectionAPI textDetectionAPI;
 
   public ImageTextDectector(TextDetectionAPI textDetectionAPI) {
     this.textDetectionAPI = textDetectionAPI;
   }
 
-  public String extractShoppingList(byte[] shoppingImageBytes)
+  public List<String> extractShoppingList(byte[] shoppingImageBytes)
       throws IOException, PhotoDetectionException {
     List<ShoppingListTextEntry> shoppingListText = textDetectionAPI.detect(shoppingImageBytes);
 
-    return createShoppingListQuery(shoppingListText);
+    return createShoppingListQueries(shoppingListText);
   }
 
-  /** Creates query from the text detected by cloudVision API. */
-  private String createShoppingListQuery(List<ShoppingListTextEntry> shoppingListText)
+  /** Creates query from the text detected by Cloud Vision API. */
+  private List<String> createShoppingListQueries(List<ShoppingListTextEntry> shoppingListText)
       throws PhotoDetectionException {
-    // ToDo: Make an algorithm to create query sentences by separating out text returned by
-    // cloudVisionAPI; to group shoppping items based on their position (y axis).
     if (shoppingListText.isEmpty()) {
       throw new PhotoDetectionException("Shopping List doesn't contain any text");
     }
 
-    String queryItem = "";
-    for (ShoppingListTextEntry singleWord : shoppingListText) {
-      queryItem += singleWord.getText() + " ";
+    // Remove the first element of this list because cloud vision API returns all the 
+    // text detetcted as the first element of the list followed by list of single 
+    // words and their properties. Examples of text returned from API can be seen in the test file.
+    shoppingListText.remove(0);
+
+    if (shoppingListText.isEmpty()) {
+      throw new PhotoDetectionException("Shopping List is empty");
+    }
+
+    // To group shoppping items based on their position.
+    String sentence = "";
+    List<String> shoppingQueries = new ArrayList<>();
+
+    // Initialize yAxisPrevLower with the current lower y value.
+    int yAxisPrevLower = shoppingListText.get(0).getLowerYBoundary();
+
+    for (ShoppingListTextEntry detectedWord : shoppingListText) {
+      int yAxisCurrentLower = detectedWord.getLowerYBoundary();
+      int yAxisCurrentUpper = detectedWord.getUpperYBoundary();
```

Comment:
There is no need to create variables for these two values. I don't think it's more readable this way and each of the variables is only used once.